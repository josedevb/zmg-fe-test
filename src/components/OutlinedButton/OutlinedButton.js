import React from 'react';
import PropTypes from 'prop-types';

import { StyledButton } from './OutlinedButton.styled';

const OutlinedButton = ({ text }) => <StyledButton>{text}</StyledButton>;

OutlinedButton.propTypes = {
  text: PropTypes.string.isRequired
};

export default OutlinedButton;
