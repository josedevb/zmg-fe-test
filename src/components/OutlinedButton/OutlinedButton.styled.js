import styled from 'styled-components';

export const StyledButton = styled.button`
  cursor: pointer;
  background-color: transparent;
  border: 1px solid ${({ theme }) => theme.colors.border.secondary};
  padding: ${({ theme }) => theme.spacing(3, 5)};
  color: ${({ theme }) => theme.colors.font.secondary};
  font-weight: bold;
  :hover {
    color: ${({ theme }) => theme.colors.font.primary};
    background-color: ${({ theme }) => theme.colors.bg.secondary};
  }
`;

export default StyledButton;
