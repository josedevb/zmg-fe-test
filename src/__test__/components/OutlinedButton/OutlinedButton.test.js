import { mount } from 'enzyme';
import { ThemeProvider } from 'styled-components';

import lightTheme from '~/styles/themes/light';
import OutlinedButton from '~/components/OutlinedButton/OutlinedButton';

describe('Test OutlinedButton', () => {
  it('Should render snapshot', () => {
    const component = mount(
      <ThemeProvider theme={lightTheme}>
        <OutlinedButton text='Hello' />
      </ThemeProvider>
    );
    expect(component).toMatchSnapshot();
  });

  it('Should call onClick', () => {
    const spy = jest.fn();
    const component = mount(
      <ThemeProvider theme={lightTheme}>
        <OutlinedButton id='outlined-id' onClick={spy} text='Hello' />
      </ThemeProvider>
    );
    const button = component.find('button');
    button.simulate('click');
    expect(spy).toHaveBeenCalled();
  });
});
