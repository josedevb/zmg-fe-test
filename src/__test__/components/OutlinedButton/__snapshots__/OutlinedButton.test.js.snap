// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test OutlinedButton Should render snapshot 1`] = `
<Component
  theme={
    Object {
      "bg": Object {
        "crystalBlack": "rgba(0, 0, 0, .18)",
        "crystalWhite": "rgba(255, 255, 255, .4)",
      },
      "blur": Object {
        "black": "rgba(0, 0, 0, .3)",
      },
      "colors": Object {
        "bg": Object {
          "badVote": "#FFAD1D",
          "badVoteCrystal": "rgba(255, 173, 29, .7)",
          "banner": "rgba(255, 255, 255, 0.8)",
          "goodVote": "#1DBBB4",
          "goodVoteCrystal": "rgba(29, 187, 180, .7)",
          "lightGray": "#EBEBEB",
          "primary": "black",
          "secondary": "white",
        },
        "border": Object {
          "primary": "black",
          "secondary": "white",
        },
        "font": Object {
          "primary": "black",
          "secondary": "white",
        },
      },
      "containerWidth": Object {
        "desktop": Object {
          "maxWidth": "50%",
          "width": "50%",
        },
        "fluid": Object {
          "maxWidth": "100%",
          "width": "100%",
        },
        "limitFluid": Object {
          "maxWidth": "1920px",
          "width": "100%",
        },
        "mobileM": Object {
          "maxWidth": "92%",
          "width": "92%",
        },
        "mobileS": Object {
          "maxWidth": "92%",
          "width": "92%",
        },
        "screen": Object {
          "maxWidth": "75%",
          "width": "75%",
        },
        "tablet": Object {
          "maxWidth": "85%",
          "width": "85%",
        },
      },
      "devices": Object {
        "desktop": "@media only screen and (min-width: 1920px)",
        "laptop": "@media only screen and (min-width: 1024px)",
        "laptopL": "@media only screen and (min-width: 1440px)",
        "mobileL": "@media only screen and (min-width: 425px)",
        "mobileM": "@media only screen and (min-width: 375px)",
        "mobileS": "@media only screen and (min-width: 320px)",
        "tablet": "@media only screen and (min-width: 768px)",
      },
      "elevation": Object {
        "backward": -1,
        "fixed": 2,
        "forward": 1,
        "higher": 3,
        "initial": 0,
        "super": 100,
      },
      "font": Object {
        "family": Object {
          "primary": "'Lato', sans-serif",
        },
        "size": Object {
          "caption": "0.75rem",
          "h1": "3.75rem",
          "h2": "3rem",
          "h3": "2.5rem",
          "h4": "2.25rem",
          "h5": "1.625rem",
          "h6": "1.5rem",
          "help": "0.625em",
          "paragraph1": "1.125rem",
          "paragraph2": "1rem",
          "paragraph3": "0.875rem",
          "small": "0.5rem",
          "subtitle1": "1.1375rem",
          "subtitle2": "1.25rem",
          "subtitle3": "1.375rem",
        },
      },
      "gradient": Object {
        "lightBlack": "rgba(0, 0, 0, .7)",
        "softGray": "rgba(47, 47, 47, .8)",
      },
      "name": "LIGHT",
      "spacing": [Function],
      "votes": Object {
        "dislike": "rgba(255, 173, 29, .8)",
        "like": "rgba(28, 187, 180, .8)",
      },
    }
  }
>
  <OutlinedButton
    inverseColors={false}
    text="Hello"
  >
    <OutlinedButtonstyled__StyledButton
      inverseColors={false}
    >
      <button
        className="OutlinedButtonstyled__StyledButton-d34t0z-0 awHVv"
      >
        Hello
      </button>
    </OutlinedButtonstyled__StyledButton>
  </OutlinedButton>
</Component>
`;
